### PR TITLE
Fix dangling relations at Notes

### PR DESCRIPTION
**Pull request by liwde**
see issue: #218 

### Visible/Frontend Changes
- [x] No more cryptic error messages ("missing(undefined)") from pouchdb
- [x] child-block.component: Will not be displayed if child was deleted from the database
- [x] child-select.component: Will lose selection if preselected child was deleted from the database

### Architectural/Backend Changes
- [x] Introduce an option to make Alerts silent, so they won't be displayed at the UI

### CODE REVIEW

File: src/app/alerts/alert.service.ts [hunk 1]
```diff
@@ -33,7 +33,9 @@ export class AlertService {
 
   addAlert(alert: Alert) {
     this.alerts.push(alert);
-    this.openSnackBar(alert);
+    if (!alert.silent) {
+      this.openSnackBar(alert);
```
Comment (sleidig) [inline src/app/alerts/alert.service.ts]: I am submitting a commit, refactoring the silent and the switch-case to decide the display of alerts into one system that's more explicit than the original `openSnackBar()` function

File: src/app/database/pouch-database.ts [hunk 2]
```diff
@@ -107,7 +107,7 @@ export class PouchDatabase extends Database {
 
 
   private notifyError(err) {
-    this.alertService.addWarning(err.message + '(' + err.id + ')');
+    this.alertService.addWarning(err.message + ' (' + err.status + ')', true);
```
Comment (sleidig) [inline src/app/database/pouch-database.ts]: If we silence the error reporting here, we should make sure that higher level modules consistently do report what fails. But I guess that's the whole point of this PR anyway.

File: src/app/children/child-block/child-block.component.ts [hunk 3]
```diff
@@ -22,8 +22,8 @@ export class ChildBlockComponent implements OnInit {
     if (this.entityId !== undefined) {
       this.entityMapper.load(Child, this.entityId).then(child => {
         this.entity = child;
-      }).catch(err => {
-        console.log(err);
+      }).catch(() => {
+        // No special error handling here, as the database will report the technical error and the UI catches the entity being undefined
```
Comment (sleidig) [inline src/app/children/child-block/child-block.component.ts]: As the database errors are silenced for the user, maybe we should display a (user-friendly) alert here?

Comment (liwde) [reply src/app/children/child-block/child-block.component.ts]: I thought about this, but just handling it silently then seemed like the better option. I hope, you can understand my reasoning (I found it very hard to put it in words):

Imagine this: Someone added a Note about some children. One of these children was then deleted. You could now (in theory) display a message whenever this child's child-block is deleted, i.e. at the Notes Overview. However, there is nothing the end user can do about it except editing the Note and then saving it (as the child-select component won't preselect children that do not exist).

Or you can hide the child-block altogether (which is what is happening now anyways, as we have nothing we could show (neither name nor picture, we have just the (technical) ID)), and don't display a message. When the Note is edited, the problem will solve itself as described above.

Once the child is deleted, there is nothing the user can do other than accepting this -- so why bother him/her? It is, however, important, to inform the user about the consequences of deleting a child in the first place -- but that's a different topic.

**Comment by liwde** [comment]
I really like your changes, the alert-display is an elegant way of handling this.

File: src/app/children/child-block/child-block.component.ts (hunk 3 above)
Comment (sleidig) [reply src/app/children/child-block/child-block.component.ts]: Yes, I get your point. That's true.
I was thinking along something like: The user reads a note talking about a child and wonders why that child is not referenced. Without a notification the behaviour of the UI is not transparent. But it's probably even more important that - as you point out - the user can't do anything about it anyway.

At the core of this is #220 anyway: this error here should be a fringe case hardly ever occurring.

Go ahead and merge!

**Pull request merged**